dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,244 @@
+# The CI Service in Hail Batch
+
+Hail Batch includes a CI service which has three functional purposes:
+
+- Runs tests against pull requests
+- Merges PRs to the `main` branch
+- Deploys services to the live infrastructure
+
+## Structure and Operation
+
+The CI service itself is deployed as a Kubernetes service in the Hail Batch cluster. See 
+[architecture diagram](../Hail%20Batch%20Architectural%20Diagram.png).
+
+As part of its configuration, CI must be configured with an access token allowing it to operate on 
+behalf of a github account called hail-ci-robot.
+
+### CI Update Loop
+
+The core CI update loop operates over a set of "watched branches" which each track three types of state flag which can 
+be marked as dirty:
+- The state of github ("`github_changed`")
+- The state of the deployed system ("`batch_changed`")
+- The state of the watched branch itself (for example - its current commit) ("`state_changed`")
+
+In hail-is, there is exactly one watched branch, which is `hail-is/hail:main`.
```

Comment:
I think it would make more sense to describe the deployment model before mentioning the organisation. Reading this, i was like "what does hail-is have to do with this". Alternatively, say something along the lines of "the default-namespace deployment of ci currently tracks `hail-is/hail:main`".

Reply:
FWIW, I don't particularly like documenting the current values of config variables as these go stale quickly. I'd link to the file where interested readers can find current values.

Reply:
Hmm, I can move it and reword it. I want to strike a balance between "documenting something very general purpose" and "explaining how it is being used". I'll try again

Reply:
I agree. This document will be useful for understanding how our CI deployment works on `hail-is/hail::main`. I think whether some behavior is hard-coded or controlled by configuration is of secondary importance. If some config variable changes frequently, then agreed we should avoid talking about its current setting, but I don't think that's the case here.